src/Tiny_httpd.ml

```diff
@@ -301,9 +301,19 @@ end
 
 module Headers = struct
   type t = (string * string) list
-  let contains = List.mem_assoc
-  let get ?(f=fun x->x) x h = try Some (List.assoc x h |> f) with Not_found -> None
-  let remove x h = List.filter (fun (k,_) -> k<>x) h
+  let contains name headers =
```

Comment:
it would make sense to store the headers as lowercase already, then, right? Directly during parsing. This way only the query string needs to be normalized for querying?

Reply:
Hmm, I was thinking it might be nicer to not change a user's input (coule be useful to maintain the input format for logging etc).

Reply:
only the keys need to be normalized, I think?

Reply:
That is correct. I guess it should be fine to re-write the keys. There shouldn't be an expectation to maintain the case of the keys. I'll update the pull request.